### DISCUSSION TITLE
RHF with Yup | set error programatically

### DISCUSSION THREAD

**Discussion by goel-ravish** [react-hook-form/react-hook-form#5759] [Q&A]
I have a form where a yup schema is passed to the resolver to validate the form fields **on blur**

There is a form field (an ID basically) that is supposed to be cross checked against a data base if it exists or not. So on blur, we are making a GET request to check the status of the ID. If it doesn't exist then we have to set an error on the field.

**Questions**
1. The validation is triggered on all the form fields simultaneously, rather than just the one which has been blurred. Is this the expected behaviour? Or my implementation is not correct?

2. Is there a way to set an error on the form field programatically?

**Comment by jorisre** [comment]
Hi 👋🏻 

Can you share a codesandbox with your issue please?

**Reply by goel-ravish** [reply]
Hi, Actually I am working on a vdi where github is blocked and copying files to the local machine is not possible. :(

**Comment by vuki656** [comment]
You can use `trigger` to only trigger validation on some fields.

https://react-hook-form.com/api/useform/trigger

Let me know if you need an example.
